routers/api/v1/utils/git.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package utils
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/git"
+)
+
+// ResolveRefOrSha resolve ref to sha if exist
+func ResolveRefOrSha(ctx *context.APIContext, ref string) string {
+	if len(ref) == 0 {
+		ctx.Error(http.StatusBadRequest, "ref not given", nil)
+		return ""
+	}
+
+	// Search branches and tags
+	for _, refType := range []string{"heads", "tags"} {
```

Comment:
Now a branch named deadbeef will override the sha deadbeef. I'm not sure if that's completely intentional, it may be best to provide some marker or mechanism to force sha over branches/tags

Reply:
well this is how it works now for status**es** and changing it would be to much for this **bugfix**

Reply:
-> #16411